### ISSUE TITLE
[DOC] using `intersphinx` - suggestion by romanlutz

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4204]
In https://github.com/sktime/sktime/issues/4189, @romanlutz suggested we could start using intersphinx in documentation for cross-linking of code references.

I think this is a good idea but also some work - comments (and contributions) welcome.

**Comment by romanlutz** [comment]
Indeed! I will add that it can be done in pieces, though. For example, if you turn on intersphinx and only change the class/function references of one file then all the others will still look exactly as they did before.

**Comment by fkiraly** [comment]
Much appreciated!

**Comment by mateuszkasprowicz** [comment]
We already use intersphinx - should we close or make the title more specific? for example, see: https://github.com/sktime/sktime/pull/5016
